`cargo doc` fails when docsrs is enabled but fips is disabled due to `cfg(any))`

https://github.com/rustls/rustls/blame/1177a465680cfac8c2a4b7217758d488d5d840c4/rustls/src/crypto/mod.rs#L571

Hi, we're trying to run cargo doc against the dioxus repo but it's failing since there's an invalid configuration on rustls/src/crypto/mod.rs:571.

The cfg takes ANY docsrs or fips. But in our repo we explicitly have fips/aws-lc-sys disabled since aws-lc-sys is causing us massive headaches (nasm requirement on windows, issues on android, etc).

```
error[E0433]: failed to resolve: use of undeclared crate or module `aws_lc_rs`
   --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustls-0.23.12/src/crypto/mod.rs:574:5
    |
574 |     aws_lc_rs::default_provider()
    |     ^^^^^^^^^ use of undeclared crate or module `aws_lc_rs`

    Checking exr v1.72.0
   Compiling built v0.7.3
For more information about this error, try `rustc --explain E0433`.
error: could not compile `rustls` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
```

The solution here is probably to provide cfg out the inner part of the function with an `unimplemented!()` if fips is disabled.

I'd be happy to make that PR but the core team might be faster.

--- a/rustls/src/crypto/mod.rs
+++ b/rustls/src/crypto/mod.rs
@@ -568,7 +568,7 @@ impl From<&[u8]> for SharedSecret {
 ///     .with_no_client_auth();
 /// # }
 /// ```
-#[cfg(any(feature = "fips", docsrs))]
+#[cfg(all(feature = "aws_lc_rs", any(feature = "fips", docsrs)))]
 #[cfg_attr(docsrs, doc(cfg(feature = "fips")))]
 pub fn default_fips_provider() -> CryptoProvider {
     aws_lc_rs::default_provider()

